### PR TITLE
fix empty filter in pattern

### PR DESCRIPTION
**Pull request by talend-sromain**
Send is empty instead of equal ""

### CODE REVIEW

**Review by jlame — APPROVED** [review]

File: src/services/filter/model/pattern-filter.model.spec.js [hunk 1]
```diff
@@ -14,306 +14,306 @@ import { FILTER_TYPE } from './filter-const.js';
...
+
+			//then
+			expect(result instanceof PatternFilter).toBeTruthy();
+			expect(result.options.values.length).toBe(1);
+			expect(result.options.values[0]).toBe("toto");
+		}));
+	});
+
+	describe('DSL actions', () => {
+		it('return filter in tql form', inject(function () {
```
Comment (jlame) [inline src/services/filter/model/pattern-filter.model.spec.js:306]: add a test for the empty case ?

File: src/services/filter/model/pattern-filter.model.js [hunk 2]
```diff
@@ -26,8 +26,17 @@ export default class PatternFilter extends ScFilter {
...
-			this.fieldId + " complies '" + value + "'";
+			const isLastIndex = index !== this.options.values.length - 1;
+
+			if (value === '') {
+				valueToString += " " + this.fieldId + " is empty ";
+			} else {
+				valueToString += " " + this.fieldId + " complies '" + value + "'";
+			}
+
+			if (isLastIndex) {
```
Comment (jlame) [inline src/services/filter/model/pattern-filter.model.js]: this variable name should be isNotLastIndex

**Review by jlame — APPROVED** [review]

**Review by jlame — CHANGES_REQUESTED** [review]
isLastIndex should be isNotLastIndex

**Review by jlame — APPROVED** [review]
